BHoM_UI/Components/Engine/SetProperty.cs

```diff
@@ -108,6 +109,16 @@ protected override void CompileInputGetters()
                 base.CompileInputGetters();
         }
 
+        /*************************************/
+
+        public override object Run(object[] inputs)
+        {
+            if (inputs != null && inputs.Length >= 1 && inputs[0] != null)
+            {
+                inputs[0] = inputs[0].DeepClone();
```

Comment:
Maybe add a comment
```cs
// Deepclone must be done before the properties are set to ensure immutability
```
and something else to explain why it has to be done at the UI level instead than in the Engine perhaps?

Reply:
Yeah - good idea

Reply:
LGTM. Agreed with @alelom